src/Serilog.Sinks.AzureDataExplorer/AzureDataExplorerSinkOptions.cs

```diff
@@ -60,14 +56,60 @@ public int QueueSizeLimit
         /// The explicit columns mapping to use for the ingested data
         /// </summary>
         public IEnumerable<SinkColumnMapping> ColumnsMapping { get; set; }
-        
+
         public IFormatProvider FormatProvider { get; set; }
 
         /// <summary>
         /// Whether to use streaming ingestion (reduced latency, at the cost of reduced throughput) or queued ingestion (increased latency, but much higher throughput).
         /// </summary>
         public bool UseStreamingIngestion { get; set; }
 
+        /// <summary>
+        /// Enables the durable mode. when specified, the logs are written to the bufferFileName first and then ingested to ADX
+        /// </summary>
+        public string BufferBaseFileName { get; set; }
+
+        /// <summary>
+        /// specifies the output format for produced logs to be written to buffer file
+        /// </summary>
+        public string BufferFileOutputFormat { get; set; }
+
+        /// <summary>
+        /// The interval at which buffer log files will roll over to a new file. The default is <see cref="RollingInterval.Hour"/>.
+        /// </summary>
+        public RollingInterval BufferFileRollingInterval { get; set; }
+
+        /// <summary>
+        /// The interval between checking the buffer files.
+        /// </summary>
+        public TimeSpan? BufferLogShippingInterval { get; set; }
+
+        ///<summary>
+        /// The maximum length of a an event record to be sent. Defaults to: null (No Limit) only used in file buffer mode
+        /// </summary>
+        public long? SingleEventSizePostingLimit { get; set; }
+
+        /// <summary>
+        /// The maximum size, in bytes, to which the buffer log file for a specific date will be allowed to grow. By default 100L * 1024 * 1024 will be applied.
+        /// </summary>
+        public long? BufferFileSizeLimitBytes { get; set; }
+
+        /// <summary>
+        /// A switch allowing the pass-through minimum level to be changed at runtime. 
+        /// </summary>
+        public LoggingLevelSwitch BufferFileLoggingLevelSwitch { get; set; }
+
+        /// <summary>
+        /// The maximum number of log files that will be retained,
+        /// including the current log file. For unlimited retention, pass null. The default is 31.
+        /// </summary>
+        public int? BufferFileCountLimit { get; set; }
+
+        /// <summary>
+        /// This property determines whether it is needed to flush the data immediately to ADX cluster,
+        /// The default is false.
+        /// </summary>
+        public bool FlushImmediately { get; set; }
 
         public AuthenticationMode AuthenticationMode { get; private set; }
         public string UserToken { get; private set; }
```

Comment:
Since we support managed identity , we need to provide an option to 

a) Find if we have to use WithSystemManagedIdentity or WithUserManagedIdentity options. Which switch determines that ? For example we say if you set the "client id" to System , then use System-MSI or user MSI

Reply:
added methods and implementations of both